lib/reporters.js

```diff
@@ -57,14 +57,34 @@ DataDogReporter.prototype.report = function(series, onSuccess, onError) {
 
     if (debug.enabled) {
         // Only call stringify when debugging.
-        debug('Calling add_metrics with %s', JSON.stringify(series));
+        debug('Calling report with %s', JSON.stringify(series));
     }
 
-    callbackify(() => metricsApi.submitMetrics({
-        body: {
-            series
+    // Distributions must be submitted via a different method than other
+    // metrics, so split them up.
+    const metrics = [];
+    const distributions = [];
+    for (const metric of series) {
+        if (metric.type === 'distribution') {
+            distributions.push(metric);
+        } else {
+            metrics.push(metric);
         }
-    }))(callback);
+    }
```

Comment:
I didn’t notice any other uses of `for...of`, but most engines added support for it at a similar time as `const`, which is used liberally. I can use a traditional `for` loop instead if this isn’t OK, or I can do another PR to update all the other traditional `for` loops to `for...of`.

Reply:
This seems fine and is cleaner than most other formats, imo.